pkg/controller/networkpolicy/store/group.go

```diff
@@ -128,6 +137,28 @@ func NewGroupStore() storage.Interface {
 			}
 			return []string{g.Selector.Namespace}, nil
 		},
+		GroupNameIndex: func(obj interface{}) ([]string, error) {
```

Comment:
Do we need this index? If we search a clustergroup by name, it can just use key by setting namespace as empty?

Reply:
We cannot use that for query. The group key is UID instead of namespace/name for internal groups

Reply:
OK, I didn't notice that. All other APIs in Antrea and K8s are using name as key and name needs to be unique. Is there any reason we do it differently for group? I'm not sure if there will be problems because of this in future. For example, some K8s lib uses the namespace/name or name as key when caching the resources or generating events, the generated code assumes name is the key. @abhiraut @Dyanngg

Reply:
I'll let @abhiraut comment but I would think that we use uid as key because the internal Group struct will be shared between clustergroups and namespaced groups, just like our internal NP case.

Reply:
i followed the internal NetworkPolicy keyfunc and as Yang mentioned above ^

Reply:
going back through the history, here's my proposal, followed by context
We change the KeyFunc to NamespacedName key func, such that the key is unique for CG and Namespaced Group(when we introduce it). This will ensure uniqueness.

The context behind NetworkPolicy changing to UID (but set in Name field) was that both ANP and K8sNP could end up with the same key (ns/name) as they could share a name. This however is not the case for Groups and I think we can safely move to using the above proposed keyFunc. @tnqn thoughts?

Reply:
https://github.com/vmware-tanzu/antrea/pull/1793

Reply:
#1793 is opened to resolve this

Reply:
I meant we'd better use the name "field" as the key to follow the K8s convention in case any generated code, clientset, apiserver logic is broken, but the name "field" could just be same as UID to avoid collision, like we did for internal NP. 
`internalNetworkPolicyKeyFunc` is used to get the key for internalNetworkPolicy from its source NetworkPolicy, it returns the UID of the source NetworkPolicy but the returned value is then set to the name field of internalNetworkPolicy. And for storages, all keyFuncs return the name field.

After looking at #1793, I think we are the same page. The proposal to use namespaced name sound good to me.